text/string.go

```diff
@@ -21,12 +25,48 @@ var (
 	rwCondition = runewidth.NewCondition()
 )
 
+type EscKind int
```

Comment:
Can you make `EscKind` and `EscSeq` private to this package? They don't need to be public.